db/upgrade.php

```diff
@@ -159,5 +159,29 @@ function xmldb_tool_excimer_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2021122400, 'tool', 'excimer');
     }
 
+    if ($oldversion < 2021122900) {
+
+        // Changing type of field flamedatad3 on table tool_excimer_profiles to binary.
+        $table = new xmldb_table('tool_excimer_profiles');
+        $field = new xmldb_field('flamedatad3', XMLDB_TYPE_BINARY, null, null, null, null, null, 'numsamples');
+
+        // A text field cannot be directly converted into a byte field.
+        $records = $DB->get_records('tool_excimer_profiles', null, '', 'id, flamedatad3');
+
+        // Launch change of type for field flamedatad3.
+        $dbman->drop_field($table, $field);
+        $dbman->add_field($table, $field);
```

Comment:
To avoid issues like the one I mentioned earlier in this PR ... `ERROR:  invalid byte sequence for encoding "UTF8"...`

I would have tried adding a new field for this data (e.g. flamedatad3gz), pointed the inserts for new profiles at this new flamedatad3 field, probably converted the existing profiles similar to how you have done here, as you still want to drop the column afterwards.

That said, it probably happened because of the state of my env as opposed to it being something that should be changed here.

Reply:
I appreciate the comment, but I stand by the way I have done it here.